armi/reactor/assemblies.py

```diff
@@ -42,15 +43,15 @@ class Assembly(composites.Composite):
     Append blocks to add them up. Index blocks with 0 being the bottom.
     """
 
+    _BLOCK_TYPE: ClassVar[Type[blocks.Block]] = None
```

Comment:
If we're allowing `self._BLOCK_TYPE` to be `None`, this helps codify that
```suggestion
    _BLOCK_TYPE: ClassVar[Optional[Type[blocks.Block]]] = None
```